packages/ag-charts-enterprise/src/features/zoom/zoom.ts

```diff
@@ -186,6 +186,7 @@ export class Zoom extends _ModuleSupport.BaseModuleInstance implements _ModuleSu
 
     private onDoubleClick(event: _ModuleSupport.InteractionEvent<'dblclick'>) {
         if (!this.enabled || !this.enableDoubleClickToReset) return;
+        if (this.ctx.interactionManager.state !== InteractionState.Default) return;
```

Comment:
Nit: Could we move these guards (which seem to be a common pattern) back into the listener config? This would also remove the direct coupling to `InteractionManager.state`.

Reply:
Another option is to expose a `checkState(state: InteractionState)` and/or `isDefaultState()`

Reply:
We could move these guards into the interaction manager. That is one of the approaches that was talking about on Tuesday. What I had in mind was some kind of state metadata that was be used to determine which interactions are allowed.

E.g.: (pseudo-code)

**animationManager.ts**
```typescript
interactionManager.pushState({
  stateName: 'animation',
  contextMenuEnabled: false,
  highlightAndTooltipsEnabled: false,
  dragEnabled: false,
  legendInteractionEnabled: false,
});
/**  run the animation */
interactionManager.popState('animation')
```

**contextMenu.ts**
```typescript
interactionManager.pushState({
  stateName: 'context-menu',
  contextMenuEnabled: true,
  highlightAndTooltipsEnable: false,
  dragEnabled: false,
  legendInteractionEnable: false,
});
/** wait for user to dismiss context-menu */
interactionManager.popState('state-menu')
```

But I think that this adds too much complexity and isn't very different to the mechanism that we've currently got. 

Event listener should not know about other events listeners, but it makes it difficult to coordinate things. I still believe that the best approach is to have individual listeners determine whether or not an event.

Perhaps a better option to reduce code duplication is to add a utility function for the `if (state === thing)` logic. e.g.:

```typescript
export function addStateConditionedEventListener(
  interactionManager: InteractionManager,
  eventType: string,
  eventHandler: (e) => any,
  requiredState: InteractionState[]
) {
  const wrapperFn = (e) => {
    if (requiredState.contains( interactionManage.state ) {
      return eventHandler(e);
    }
  }
  interactionManager.addEventListener( eventType, wrapperFn );
```

Reply:
I like your last proposal, and continuing our conversation about the state enum it could be a great place to use these flags (instead of an array) to support multiple states at once.

```
export enum InteractionState {
    Default = 1,
    ZoomPan = 2,
    ContextMenu = 4,
    Animation = 8,
    All = 15
}
// ...
addEventListener(eventName: string, eventHandler: (...args) => void, eventState = InteractionState.Default) // or InteractionState.All
// ...
interactionManager.addEventListener( eventType, (e) => {
  if (this.state & eventState) {
    eventHandler(e);
  }
});
```

Reply:
Okay. SGTM 👍

Reply:
Done.